src/codegen/compile.ml

```diff
@@ -10355,6 +10355,14 @@ and compile_const_decs env pre_ae decs : (VarEnv.t -> VarEnv.t) * (E.t -> VarEnv
         (fun env ae -> fill1 env ae; fill2 env ae) in
   go pre_ae decs
 
+and const_exp_matches_pat env pat exp : bool =
```

Comment:
`pat_matches_const_exp`? Not sure who matches who…

Reply:
Maybe @crusso can suggest a better naming.